python/lsst/analysis/drp/colorColorFitPlot.py

```diff
@@ -0,0 +1,385 @@
+                                  pipelineConnections=ColorColorFitPlotTaskConnections):
+
+    axisActions = ConfigurableActionStructField(
+        doc="The actions to use to calculate the values on each axis.",
+        default={"xAction": MagDiff, "yAction": MagDiff, "magAction": MagColumnNanoJansky},
+    )
+
+    axisLabels = pexConfig.DictField(
+        doc="Axis labels for the plot.",
+        keytype=str,
+        itemtype=str,
+    )
+
+    sourceSelectorActions = ConfigurableActionStructField(
+        doc="What types of sources to use.",
+        default={"sourceSelector": dataSelectors.StarIdentifier},
+    )
+
+    selectorActions = ConfigurableActionStructField(
+        doc="Which selectors to use to narrow down the data for QA plotting.",
+        default={"flagSelector": dataSelectors.CoaddPlotFlagSelector},
+    )
+
+    stellarLocusFitDict = pexConfig.DictField(
+        doc="The parameters to use for the stellar locus fit.",
+        keytype=str,
+        itemtype=float,
+    )
+
+    fitName = pexConfig.Field(
+        doc="The name of the fit.",
+        dtype=str,
+    )
+
+
+class ColorColorFitPlotTask(pipeBase.PipelineTask):
+
+    ConfigClass = ColorColorFitPlotTaskConfig
+    _DefaultName = "ColorColorFitPlotTask"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # Docs inherited from base class
+
+        columnNames = set(["patchId"])
+        for actionStruct in [self.config.axisActions, self.config.selectorActions,
+                             self.config.sourceSelectorActions]:
+            for action in actionStruct:
+                for col in action.columns:
+                    columnNames.add(col)
+
+        inputs = butlerQC.get(inputRefs)
+        dataFrame = inputs["catPlot"].get(parameters={"columns": columnNames})
+        inputs["catPlot"] = dataFrame
+        dataId = butlerQC.quantum.dataId
+        inputs["dataId"] = dataId
+        inputs["runName"] = inputRefs.catPlot.datasetRef.run
+        localConnections = self.config.ConnectionsClass(config=self.config)
+        inputs["tableName"] = localConnections.catPlot.name
+        inputs["fitParams"] = self.config.stellarLocusFitDict
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+        plt.close()
+
+    def run(self, catPlot, dataId, runName, tableName, fitParams):
+
+        mask = np.ones(len(catPlot), dtype=bool)
+        for actionStruct in [self.config.selectorActions, self.config.sourceSelectorActions]:
+            for selector in actionStruct:
+                mask &= selector(catPlot)
+        catPlot = catPlot[mask]
+
+        columns = {self.config.axisLabels["x"]: self.config.axisActions.xAction(catPlot),
+                   self.config.axisLabels["y"]: self.config.axisActions.yAction(catPlot),
+                   self.config.axisLabels["mag"]: self.config.axisActions.magAction(catPlot),
+                   "patchId": catPlot["patchId"]}
```

Comment:
I don't understand why patchId is being passed here. Why is this needed in the `runQuantum` method?